apps/explorer/lib/explorer/chain/optimism/withdrawal.ex

```diff
@@ -274,22 +274,48 @@ defmodule Explorer.Chain.Optimism.Withdrawal do
     withdrawal_l2_block_number <= last_root_l2_block_number
   end
 
-  defp game_by_index(game_index) do
-    if not is_nil(game_index) do
-      Repo.replica().one(
-        from(
-          g in DisputeGame,
-          select: %{created_at: g.created_at, resolved_at: g.resolved_at, status: g.status},
-          where: g.index == ^game_index
-        )
+  # Fetches dispute game info from DB by game's contract address or game's unique index.
+  #
+  # ## Parameters
+  # - `game_address_hash`: Game's contract address. Must be `nil` if unknown.
+  # - `game_index`: Game unique index. Must be `nil` if unknown.
+  #
+  # ## Returns
+  # - A map with necessary info about the dispute game.
+  # - `nil` if the dispute game not found in DB or both input parameters are `nil`.
+  @spec game_by_address_hash_or_index(Hash.t() | nil, non_neg_integer() | nil) ::
+          %{:created_at => DateTime.t(), :resolved_at => DateTime.t(), :status => non_neg_integer()} | nil
+  defp game_by_address_hash_or_index(nil, nil), do: nil
+
+  defp game_by_address_hash_or_index(game_address_hash, nil) do
+    Repo.replica().one(
+      from(
+        g in DisputeGame,
+        select: %{created_at: g.created_at, resolved_at: g.resolved_at, status: g.status},
```

Comment:
This can be simplified a bit if to define fields array beforehand:

```
  @game_fields ~w(
    created_at
    resolved_at
    status
  )
```

```suggestion
        select: map(g, ^@game_fields),
```

Reply:
Done in https://github.com/blockscout/blockscout/pull/12332/commits/e91717533a084aed4760aba918e7313040e54020